src/modules/commander/Commander.cpp

```diff
@@ -1084,15 +1028,196 @@ Commander::handle_command(vehicle_status_s *status_local, const vehicle_command_
+						_worker_thread.startTask(WorkerThread::Request::ESCCalibration);
+
+					} else {
+						answer_command(cmd, vehicle_command_s::VEHICLE_CMD_RESULT_DENIED, command_ack_pub);
+					}
+
+				} else if ((int)(cmd.param4) == 0) {
+					/* RC calibration ended - have we been in one worth confirming? */
+					if (status_flags.rc_input_blocked) {
+						/* enable RC control input */
+						status_flags.rc_input_blocked = false;
+						mavlink_log_info(&mavlink_log_pub, "Calibration: Restoring RC input");
+					}
+
+					answer_command(cmd, vehicle_command_s::VEHICLE_CMD_RESULT_ACCEPTED, command_ack_pub);
+
+				} else {
+					answer_command(cmd, vehicle_command_s::VEHICLE_CMD_RESULT_UNSUPPORTED, command_ack_pub);
+				}
+			}
+
+			break;
+		}
+
+	case vehicle_command_s::VEHICLE_CMD_FIXED_MAG_CAL_YAW: {
+			// Magnetometer quick calibration using world magnetic model and known heading
+			if ((status.arming_state == vehicle_status_s::ARMING_STATE_ARMED)
+			    || (status.arming_state == vehicle_status_s::ARMING_STATE_SHUTDOWN)
+			    || _worker_thread.isBusy()) {
+
+				// reject if armed or shutting down
+				answer_command(cmd, vehicle_command_s::VEHICLE_CMD_RESULT_TEMPORARILY_REJECTED, command_ack_pub);
+
+			} else {
+				answer_command(cmd, vehicle_command_s::VEHICLE_CMD_RESULT_ACCEPTED, command_ack_pub);
+				// parameter 1: Yaw in degrees
+				// parameter 3: Latitude
+				// parameter 4: Longitude
+
+				// assume vehicle pointing north (0 degrees) if heading isn't specified
+				const float heading_radians = PX4_ISFINITE(cmd.param1) ? math::radians(roundf(cmd.param1)) : 0.f;
+
+				float latitude = NAN;
+				float longitude = NAN;
+
+				if (PX4_ISFINITE(cmd.param3) && PX4_ISFINITE(cmd.param4)) {
+					// invalid if both lat & lon are 0 (current mavlink spec)
+					if ((fabsf(cmd.param3) > 0) && (fabsf(cmd.param4) > 0)) {
+						latitude = cmd.param3;
+						longitude = cmd.param4;
+					}
+				}
+
+				status_flags.condition_calibration_enabled = true;
+				_worker_thread.setMagQuickData(heading_radians, latitude, longitude);
+				_worker_thread.startTask(WorkerThread::Request::MagCalibrationQuick);
+			}
+
+			break;
+		}
+
+	case vehicle_command_s::VEHICLE_CMD_PREFLIGHT_STORAGE: {
+
+			if ((status.arming_state == vehicle_status_s::ARMING_STATE_ARMED)
+			    || status.arming_state == vehicle_status_s::ARMING_STATE_SHUTDOWN
+			    || _worker_thread.isBusy()) {
+
+				// reject if armed or shutting down
+				answer_command(cmd, vehicle_command_s::VEHICLE_CMD_RESULT_TEMPORARILY_REJECTED, command_ack_pub);
+
+			} else {
+
+				if (((int)(cmd.param1)) == 0) {
+					answer_command(cmd, vehicle_command_s::VEHICLE_CMD_RESULT_ACCEPTED, command_ack_pub);
+					_worker_thread.startTask(WorkerThread::Request::ParamLoadDefault);
+
+				} else if (((int)(cmd.param1)) == 1) {
+					answer_command(cmd, vehicle_command_s::VEHICLE_CMD_RESULT_ACCEPTED, command_ack_pub);
+					_worker_thread.startTask(WorkerThread::Request::ParamSaveDefault);
```

Comment:
We could update all of these commands to operate within the worker in parameters (not suggesting doing it here).